tests/integrationtests/tst_perfparser.cpp

```diff
@@ -825,6 +825,68 @@ private slots:
 
         QCOMPARE(decompressed.readAll(), QByteArrayLiteral("Hello World\n"));
     }
+
+    void testArchive_data()
+    {
+        QTest::addColumn<QByteArray>("content");
+        QTest::addColumn<QString>("filename");
+
+        QTest::addRow("uncompressed tar") << qUncompress(QByteArray::fromBase64(QByteArrayLiteral(
```

Comment:
this is really ugly and opaque stuff that I feel uncomfortable with it, can we find another way?